gr-digital/include/gnuradio/digital/constellation.h

```diff
@@ -127,7 +131,7 @@ class DIGITAL_API constellation : public std::enable_shared_from_this<constellat
      *
      * This is expensive to compute.
      */
-    void gen_soft_dec_lut(int precision, float npwr = 1.0);
```

Comment:
This function does not make mathematical sense without a noise power argument. It's a key part of the calculation. Because it's an API change I've defaulted it to the previously hard coded value.

Reply:
Changing this resolved the need for the conditionals I had in the class previously that checked for the sentinel value from the soft decoder. (Previous PR).
As far as I can tell, adding noise power as an argument was a relatively new change and it was unused anyway.

Reply:
A user can achieve the same behavior as before by calling set_npwr(npwr) first and then using the LUT or soft dec functions. Both of these functions require information stored in class members anyway, so the API change will ensure their proper use and I think is better architecture.